### ISSUE TITLE
scDD warnings 

### ISSUE THREAD

**Issue by mRcSchwering** [kdkorthauer/scDD#8]
Hi,

the scDD function with `scDD(se, prior_param = prior_param, testZeroes = FALSE)` always produces a lot of these warnings: 
`In MAP1[tofit[g], ][normExprs(SCdat[tofit[g], colData(SCdat)[[condition]] ==  ... :
  number of items to replace is not a multiple of replacement length`
Is that normal?

**Comment by kdkorthauer** [comment]
Hi @mRcSchwering,

Thanks for bringing this to my attention. No, those warnings are not expected. Can you provide me with some more information so I can track down the root cause? 

Are you using the most recent version of scDD from Bioconductor? If not, could you try installing this version with the following and re-running?

```
## try http:// if https:// URLs are not supported
source("https://bioconductor.org/biocLite.R")
biocLite("scDD"))
```

If you are using the most recent version, would it be possible to send me a small portion of the `se` object you are using that throws these errors?  My email address is keegan@jimmy.harvard.edu.

Best,
Keegan

**Comment by mRcSchwering** [comment]
Hey @kdkorthauer,

Now that I tried to install it from bioconductor I realized that I had the github version.
When I installed it I also changed the description file to `Depends: R (>= 3.3)` because I did not want to install R 3.4.
I thought the package is probably not really using new functionalities from R 3.4.

My bad, I should try this again with R 3.4.

Best
Marc

**Comment by kdkorthauer** [comment]
Hi Marc,

The R version dependency is a requirement of Bioconductor (to be included in the devel release). After the next stable Bioconductor release (next month), this will no longer be an issue.

I am not sure that the issue is due to the R version itself, but it is worth a try. If you do not want to install R 3.4 on your system, you could send me a portion of the `eset` that throws the warnings, and I can check it out on my system.

Best,
Keegan

**Comment by kdkorthauer** [comment]
Hi Marc,

Thanks for sending the example object.  I was able to reproduce the problem on my system with the current version of the code (under R 3.4), and I think I determined what is causing the warnings.  The example object you sent had many values in the normalized counts matrix `assays(se)$NormCounts` that were negative (but very small, e.g.  -2.89e-06). I'm guessing that these actually represent zero (no counts detected), and that this is an artifact of the way you have normalized? `scDD` assumes that counts are all non-negative, so negative counts will cause problems. I tried changing all the negative values to exactly zero, and then the function ran without warnings. 

Now that I've tracked down the issue, I'll add a check for negative expression values and an informative error message so that this doesn't cause problems in the future.

Best,
Keegan
